static/js/components/EmploymentForm.js

```diff
@@ -20,8 +20,35 @@ import SelectField from './inputs/SelectField';
 import CountrySelectField from './inputs/CountrySelectField';
 import StateSelectField from './inputs/StateSelectField';
 import type { WorkHistoryEntry } from '../flow/profileTypes';
+import type { Validator, UIValidator } from '../util/validation';
+import type {
+  Profile,
+  BoundSaveProfile,
+  ValidationErrors,
+  BoundUpdateProfile,
+} from '../flow/profileTypes';
+import type { UIState } from '../reducers/ui';
+import type { SetWorkHistoryEdit } from '../containers/ProfileFormContainer';
 
 class EmploymentForm extends ProfileFormFields {
+  props: {
+    profile:                          Profile,
+    ui:                               UIState;
+    updateProfile:                    BoundUpdateProfile,
+    saveProfile:                      BoundSaveProfile,
+    clearProfileEdit:                 () => void,
+    errors:                           ValidationErrors,
+    setWorkDialogVisibility:          () => void,
+    setWorkDialogIndex:               () => void,
+    setWorkDegreeLevel:               () => void,
+    setWorkDegreeInclusions:          () => void,
+    setWorkHistoryEdit:               SetWorkHistoryEdit,
+    deletionIndex:                    number,
+    setShowWorkDeleteAllDialog:       (bool: boolean) => void,
```

Comment:
to me it's a bit weird that you need to give them names at all in function type declarations, but maybe it's a difficulty they ran into when designing the syntax.